readme-cn.md

```diff
@@ -0,0 +1,1799 @@
+    </a>
+    {// but we can prefetch it!
+    Router.prefetch('/dynamic')}
+  </div>
+```
+
+路由实例只允许在应用程序的客户端。以防服务端渲染发生错误，建议prefetch事件写在`componentDidMount()`生命周期里。
+
+```jsx
+import React from 'react'
+import Router from 'next/router'
+
+export default class MyLink extends React.Component {
+  componentDidMount() {
+    Router.prefetch('/dynamic')
+  }
+  
+  render() {
+    return (
+       <div>
+        <a onClick={() => setTimeout(() => url.pushTo('/dynamic'), 100)}>
+          A route transition will happen after 100ms
+        </a>
+      </div>   
+    )
+  }
+}
+```
+
+<a id="custom-server-and-routing" style="display: none"></a>
+### 自定义服务端路由
+
+<p><details>
+  <summary><b>Examples</b></summary>
+  <ul>
+    <li><a href="./examples/custom-server">Basic custom server</a></li>
+    <li><a href="./examples/custom-server-express">Express integration</a></li>
+    <li><a href="./examples/custom-server-hapi">Hapi integration</a></li>
+    <li><a href="./examples/custom-server-koa">Koa integration</a></li>
+    <li><a href="./examples/parameterized-routing">Parameterized routing</a></li>
+    <li><a href="./examples/ssr-caching">SSR caching</a></li>
+  </ul>
+</details></p>
+
+
+When using a custom server with a server file, for example called `server.js`, make sure you update the scripts key in `package.json` to:
+
+一般你使用 `next start` 命令来启动next服务，你还可以编写代码来自定义路由。
+
+```json
+{
+  "scripts": {
+    "dev": "node server.js",
+    "build": "next build",
+    "start": "NODE_ENV=production node server.js"
+  }
+}
+```
+
+下面这个例子使 `/a` 路由解析为`./pages/b`，以及`/b` 路由解析为`./pages/a`;
+
+```js
+// This file doesn't go through babel or webpack transformation.
+// Make sure the syntax and sources this file requires are compatible with the current node version you are running
+// See https://github.com/zeit/next.js/issues/1245 for discussions on Universal Webpack or universal Babel
+const { createServer } = require('http')
+const { parse } = require('url')
+const next = require('next')
+
+const dev = process.env.NODE_ENV !== 'production'
+const app = next({ dev })
+const handle = app.getRequestHandler()
+
+app.prepare().then(() => {
+  createServer((req, res) => {
+    // Be sure to pass `true` as the second argument to `url.parse`.
+    // This tells it to parse the query portion of the URL.
+    const parsedUrl = parse(req.url, true)
+    const { pathname, query } = parsedUrl
+
+    if (pathname === '/a') {
+      app.render(req, res, '/b', query)
+    } else if (pathname === '/b') {
+      app.render(req, res, '/a', query)
+    } else {
+      handle(req, res, parsedUrl)
+    }
+  }).listen(3000, err => {
+    if (err) throw err
+    console.log('> Ready on http://localhost:3000')
+  })
+})
+```
+
+`next`的API如下所示
+- `next(opts: object)`
+
+opts的属性如下:
+- `dev` (`boolean`) 判断 Next.js应用是否在开发环境 - 默认 `false`
+- `dir` (`string`) Next项目路径 - 默认 `'.'`
+- `quiet` (`boolean`) Hide error messages containing server information - 默认 `false`
```

Comment:
Not translated.

Reply:
Thank you for your suggestion, I have translated it.